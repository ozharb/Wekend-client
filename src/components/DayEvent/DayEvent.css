.Day-Events > h2{
 font-family: 'Public Sans', sans-serif;
 font-size: 50px;
}
.Day-Events > h3{
    font-family: 'Public Sans', sans-serif;

   }
.Day-Events_heading-large{
    font-size: 40px;
    margin-bottom: 5px;
}
.Day-Events_heading-small{
    font-size: 20px;
    margin-bottom: 5px;
}
.Day-Events:nth-child(even) button{
    background: black;
    color: white;
}
.Day-Events:nth-child(even) p{
    color: white;
}
.Day-Events:nth-child(even) a{
    text-decoration: none;
    padding: 10px;
    background: white;
    color: black;
    margin-top: 5px;
}
.host-name{
    font-weight: bold;
}
.Day-Events:nth-child(odd) button{
    background: white;
    color: black;
}
.Day-Events:nth-child(odd) p{
    color: black;
}
.Day-Events:nth-child(odd) a{
    text-decoration: none;
    padding: 10px;
    height: 15px;
    background: black;
    color: white;
    margin-top: 5px;
}
.Event__expand-event-button{
    width: 100%;
    margin: 0;
}
.Event-edit-container{
    float: right;

}.Event-edit-container>a{
    font-size: medium;
    margin-left: 8px;

}
.time-letters{
   font-size: 20px;
   margin-bottom: 5px;
}
.Event-Content{
  
    display: flex;

    
}
.alert-message{
    text-align: center;
    
}
.alert{
    color: red;

}

.Event-top-and-bottom{

    width: 100%;
    display: flex;
    flex-direction: column;
    align-self: flex-end;
}
.Event-content-details{

    display: flex;
    flex-direction: row;
    justify-content: space-around;
}
.events-attendees {
width: 100px;
}
.events-attendees > li{
    text-align: start;
    font-weight: bold;
    padding-right: 1px;

}

::marker {
    margin-left: 20px;
}

.Event-column-1{
 margin: 10px;
 font-size: medium;
 line-height: 1.4em; 
 margin-top: 14px;
}
.deets{
    letter-spacing: .3em;

}
.whos-going{
letter-spacing: .14em;
}

.Event-column-2{
    font-family: 'Monoton', cursive;
 font-size: 50px;
 font-size: bold;

}
.alert-on{
    color: red;
}
.Event-bottom-row{
    letter-spacing: .15em;
    font-size: 20px;
    font-weight: bold;
    margin-bottom: 20px;
}

@media screen and (min-width: 670px) {
    .Header{
    height: 42px;
    }
    .Header h1 {
    font-size: 22pt;
    }
    
    .Header__not-logged-in {
    flex-direction: row;
    }
    
    .Header__not-logged-in .Hyph {
    display: initial;
    }
    }