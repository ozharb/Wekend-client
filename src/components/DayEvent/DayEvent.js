import toast, { Toaster } from 'react-hot-toast';
import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { Link, withRouter } from 'react-router-dom'
import WekendApiService from '../../services/Wekend-api-service'

// import { Link } from 'react-router-dom'
// import { ThingStarRating } from '../ThingStarRating/ThingStarRating'
import './DayEvent.css'
import EventsContext from '../../contexts/EventsContext';


const convertTime = require('convert-time');
export default class DayEvent extends Component {
    static contextType = EventsContext
    state = {
        expand: false,
        error: null,
        alert: false
    };
    handleExpand = e => {
        this.setState({
            expand: !this.state.expand
        })
    }
    
    
    render() {  
               const { event } = this.props
              
        let username = 'user'
        const user =  window.localStorage.getItem(username)
        const attending = (event.attendees.length)
        ? (!!event.attendees.find(i=>i.username===user))
        : false
        console.log("event ATENDEES:",event.attendees.length)
        
        const rsvpLink = (attending)?`cancel-rsvp`:`rsvp`
        const path = `/${rsvpLink}/${event.id}`
         let alertStyle = event.alert ? {color: 'red'} : null
        let headingStyle = (event.title.length > 12 && this.state.expand)? 'Day-Events_heading-small' : 'Day-Events_heading-large'
        let time = convertTime(event.time)
        let timeNumbers = time.split('').slice(0, time.length - 2).join('')
        let timeLetters = time.split('').slice(time.length - 2).join('')
        
        const notify = () => toast((t) => (
            <span className = 'alert-message'>
               <b>{event.Event_Host===user?`You`: event.Event_Host}</b>
        {'  '}changed this event.
        <br/>
              <button onClick={() => toast.dismiss(t.id)}>
                Cool
              </button>
            </span>
       
          ) 
             ,console.log('alert')
              ,WekendApiService.turnOffAlert(event.id)
         
            .then(this.context.changeEventAlert(event.id))
              .catch(error => { console.error({ error }) })
            
          );
         const host = event.Event_Host ===user ? <>
           <span className='host-name'>You're</span> hosting
         </> 
         : <>
         <span className='host-name'>{event.Event_Host}'s</span> hosting
       </> 
       
        // const expandButtonText = this.state.expand
        // ?  <i className="fas fa-chevron-down"><FontAwesomeIcon className='chevron' icon='chevron-down' /></i> 
        // :   <i className="fas fa-chevron-right"><FontAwesomeIcon className='chevron' icon='chevron-right' /></i>
        const eventDetails = this.state.expand

            ?<div className = 'Event-top-and-bottom'>
             <div className='Event-content-details'>
                <div className="Event-column-1">
                  {host} <br/>
                  @ {event.place}
                </div>
                <div className="Event-column-2">
                    {timeNumbers}
                    <div className='time-letters'>{timeLetters}</div>
                </div>
            </div>
            <div className = 'event-details'>
    <div className='Event-bottom-row'> 
    Deets:{event.details.length===0? 'No deets':event.details.split(/\n \r|\n/).map((para, i) =>
                    <p key={i}>{para}</p>)}
          </div>          
    {event.Event_Host === user 
     ?<div className='Event-edit-container'>
                                 <Link
                        to={`/change-event/${event.id}`}
                        className='Event-edit-link'>
                        edit
                     </Link>
                     <Link
                        to={{pathname:`/delete-event/${event.id}`,
                            state:{ title: event.title  }
                        }}
                        className='Event-edit-link'>
                        delete
                     </Link>
                        </div>
                     : <div className='Event-edit-container'>
                     <Link
            to={{pathname: path,
            state: {
                title: event.title
            }}}
            className='Event-edit-link'>
            {attending?`change rsvp`: `rsvp`}
         </Link>
            </div>}
                </div>
               
                </div>
            : null
            
       const alertOrExpand = event.alert? notify : this.handleExpand

        return (
            <>
                {/* <Link to={`/thing/${thing.id}`} className='ThingListItem'> */}
                <div className='Day-Events'>
          
     
      <Toaster position="top-center"/>

                    <button
                        type='button'
                        aria-label='expand'
                        onClick={alertOrExpand}
                        className='Event__expand-event-button'>
                        <div className="Event-Content">
                            <div className='event-title-attendees'>
                                <h2 className={headingStyle} style={alertStyle}>{event.title}</h2>
                                {this.state.expand && 
                                <ul className = 'events-attendees'>
                                    Who's going:
                                   { event.attendees.length === 0? <p>Nobody</p>:event.attendees.map((el,i)=>{
                                   return <li key={i+event.id}>{el.username}</li>
                                })}
                                </ul> }
                                
                                {/* <h2 className='Day-Events_heading'>{event.time}</h2> */}
                            </div>
                   
                            {eventDetails}

                        </div>
                    </button>
                </div>
                {/* </Link> */}
            </>
        )
    }
}

// function readableReviewCount(number) {
//   switch(number) {
//     case 0:
//       return 'no reviews yet'

//     case 1:
//       return `based on 1 review`

//     default:
//       return `based on ${number} reviews`
//   }
// }

// function truncate(text) {
//   const words = text.split(' ')

//   if (words.length > 10) {
//     return words.slice(0, 10).join(' ') + ' ...'
//   }

//   return text
// }
