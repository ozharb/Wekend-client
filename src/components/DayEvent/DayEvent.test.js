import React from 'react';
import DayEvent from './DayEvent';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

it('renders the UI as expected', () => {
  const tree = renderer
    
    .create(
    <BrowserRouter>
    <DayEvent 
    props={{event:{attendees:[{username:''}], title:'', time: '0:00'}}}
    event={{attendees:[{username:''}], title:'', time:'0:00'}}
    time={'0:00'}
    />
     </BrowserRouter>
    )

    
    .toJSON();
 
  expect(tree).toMatchSnapshot();  
  });