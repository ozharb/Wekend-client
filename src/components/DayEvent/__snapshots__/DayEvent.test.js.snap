// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Day-Events"
>
  <div
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "position": "fixed",
        "zIndex": 9999,
      }
    }
  />
  <section
    aria-label="expand"
    className="Event__expand-event-button"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
  >
    <div
      className="Event-Content"
    >
      <div
        className="event-title-attendees"
      >
        <h2
          className="Day-Events_heading-large"
          style={null}
        >
          
        </h2>
      </div>
    </div>
  </section>
</div>
`;
